tests/common/include/tests/common/core_detector.inl

```diff
@@ -608,10 +608,10 @@ TEST(detector, detector_volume_construction) {
               sf_finder_links[geo_obj_id::e_passive]);
 
     EXPECT_EQ(d.surfaces().size(), 19u);
-    EXPECT_EQ(d.mask_store().template size<mask_id::e_portal_cylinder2>(), 3u);
+    EXPECT_EQ(d.mask_store().template size<mask_id::e_portal_cylinder2>(), 2u);
     EXPECT_EQ(d.mask_store().template size<mask_id::e_portal_ring2>(), 4u);
     EXPECT_EQ(d.mask_store().template size<mask_id::e_annulus2>(), 3u);
-    EXPECT_EQ(d.mask_store().template size<mask_id::e_cylinder2>(), 3u);
+    EXPECT_EQ(d.mask_store().template size<mask_id::e_cylinder2>(), 1u);
```

Comment:
and also here

Reply:
Why was it `3u` before this PR?

Reply:
I think, because this is the core detector test, which is just prefilling some surfaces into the default detector type. It is not actually the toy detector we are talking about here. But the reason is the same as on the comment above, previously ```e_cylinder2``` and ```e_portal_cylinder2``` were referring to the same cylinder type and hence a single shared mask collection in the mask store. Now, the portal cylinders are split from all other cylinders, so ```e_cylinder2 != e_portal_cylinder2``` and instead of 3 cylinders in a single collection, we have two collections (one with two portal cylinders and one with a passive or active cylinders that needs a more complicated intersector)

Reply:
OK